[SofaGraphComponent] Run SceneChecker at each load

guparan opened the pull request:
As discussed in latest dev meeting, we need SceneChecker output even at first load.

______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

guparan committed:
[SofaGuiQt] Run SceneChecker on each load

guparan committed:
[SofaGraphComponent] Better output for SceneCheckUsingAlias

guparan committed:
[SofaGraphComponent] Simplify SceneCheckAPIChange

guparan commented:
[ci-build][with-scene-tests]

guparan commented:
Will try to make it run with any GUI (currently only with RealGUI).

guparan committed:
[SofaGuiQt] Remove SceneChecker mechanism from GUI

guparan committed:
[SofaSimulationCore] Delegate doLoad to force notifications

SceneLoaders (derived from SceneLoaderFactory) must now override "doLoad" as they used to override "load".
-> Calling load will call notifyLoadingSceneBefore + doLoad + notifyLoadingSceneAfter

This permits to better customize pre/post scene loading with SceneLoader::Listener

guparan committed:
[SofaGraphComponent] ADD SceneCheckerListener for SceneCheck mechanism

SceneCheckerListener is a SceneLoader::Listener that will call SceneCheckVisitor::validate on every notifyLoadingSceneAfter event.
This permits to have SceneChecking working with any GUI (classic, batch) and any SceneLoader (XML, Python).

guparan committed:
[SofaGraphComponent] Better log for SceneChecker

guparan committed:
[SofaGraphComponent] License year

guparan commented:
So I think I found a clean way to run the SceneChecks by using SceneLoader::Listener.
SceneChecks are now run on every GUI (batch, classic) and every SceneLoader (XML, Python).
They are also triggered in unit tests that use loadFromMemory but that is maybe too much...

What do you guys think?

guparan commented:
So apparently SceneLoaderPY (Python scene loader) needs to call notifyLoadingScene at a specific moment because using my new `doLoad` delegate messes with the unit test SofaPython_test/python/moduleReload.py

@damienmarchal Could you tell me if you see a fix to that?

We still can stick to the old `load` override and ignore `doLoad` in SceneLoaderPY.

damienmarchal commented:
Actually the SceneListener was specifically designed for the python modules/scripts to be cleaned/reloaded when a new scene is loaded.  This that few years ago so I forgot a bit how it work. But I'm sure it is important not to break it :)

guparan committed:
[SofaGraphComponent] MOVE SceneCheckerListener registration to runSofa

+ add runSofa option to disable scene checking: noscenecheck

guparan committed:
[SofaPython] FIX SceneLoaderPY scene loading notifications

loadSceneWithArguments did not call notifyLoadingSceneBefore and notifyLoadingSceneAfter

guparan commented:
@damienmarchal I fixed the problem with SceneLoaderPY :-)
